geps/gep-2162.md

```diff
@@ -0,0 +1,106 @@
+
+## API
+
+This GEP proposes API changes describes as follow:
+1. Update the `GatewayClassStatus` struct to include a string-represented list of `SupportedFeatures`.
+
+```go
+// GatewayClassStatus is the current status for the GatewayClass.
+type GatewayClassStatus struct {
+    // Conditions is the current status from the controller for
+    // this GatewayClass.
+    //
+    // Controllers should prefer to publish conditions using values
+    // of GatewayClassConditionType for the type of each Condition.
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=type
+    // +kubebuilder:validation:MaxItems=8
+    // +kubebuilder:default={{type: "Accepted", status: "Unknown", message: "Waiting for controller", reason: "Pending", lastTransitionTime: "1970-01-01T00:00:00Z"}}
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // SupportedFeatures is the features the GatewayClass support.
+    SupportedFeatures []string `json:"SupportedFeatures,omitempty"`
+}
+```
+
+## Conformance Details
+
+(TBD)
+
+## Alternatives
+
+### Re-using ConformanceProfiles structs
+
+We could use the same structs as we do in conformance profile object, more specifically, the [ProfileReport](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/apis/v1alpha1/profilereport.go#LL24C6-L24C19) struct.
+
+Though it would be nice to have only one place to update, these structs seems to include much more data relevant to the conformance report but not for our use case. 
+
+That said, conformance profiles are still at experimental stage, we could explore the option to create a shared struct that will be used both for the conformance reports and for the GatewayClass status.
+
+### Instruct users to read from the future conformance profiles report
+
+The current plan for conformance profiles is to also include centralized reporting. (more info in [gep-1709](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-1709.md))
+We could wait for this to be implemented and instruct users to read from that source to determine what features their installed GatewayClass support.
+
+However, having the supported features published in the GatewayClass Status adds the following values:
+* We could build a mechanism or a tool to block or warn when unsupported features are used.
+* Users will be able to select the GatewayClass that suits their needs without having to refer to documentation or conformance reports.
+
+This does not cover a future piece of work we want to implement which is to warn/block users from applying a Gateway API object if the installed GWC doesn't support it. (originally suggested in [#1804](https://github.com/kubernetes-sigs/gateway-api/issues/1804)). 
+
+
+## References
+
+https://github.com/kubernetes-sigs/gateway-api/discussions/2108
+https://github.com/kubernetes-sigs/gateway-api/issues/1804
```

Comment:
Can you turn these into links? (Our markdown parser does not do that automatically).

Reply:
I looked at gep 1709 https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-1709.md?plain=1#L631C11-L632

The reference bit does look the same, what am I missing?

Reply:
Looks like you found another bug 🙃: https://gateway-api.sigs.k8s.io/geps/gep-1709/#references